### PR TITLE
Nifi-ldap-iaa support for PasswordComparisonAuthenticator

### PR DESCRIPTION
**Pull request by felixalbani**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by alopresto** [comment]
Will review.

File: nifi-nar-bundles/nifi-ldap-iaa-providers-bundle/nifi-ldap-iaa-providers/src/main/java/org/apache/nifi/ldap/LdapProvider.java [hunk 1]
```diff
@@ -38,11 +38,10 @@
 import org.springframework.ldap.core.support.SimpleDirContextAuthenticationStrategy;
 import org.springframework.security.authentication.BadCredentialsException;
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.authentication.encoding.LdapShaPasswordEncoder;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
-import org.springframework.security.ldap.authentication.AbstractLdapAuthenticationProvider;
-import org.springframework.security.ldap.authentication.BindAuthenticator;
-import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.security.ldap.authentication.*;
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-ldap-iaa-providers-bundle/nifi-ldap-iaa-providers/src/main/java/org/apache/nifi/ldap/LdapProvider.java]: I imagine this was your IDE collapsing these imports, but our [coding style guidelines](https://cwiki.apache.org/confluence/display/NIFI/Contributor+Guide#ContributorGuide-CodeStyle) discourage the use of wildcard imports.

File: nifi-nar-bundles/nifi-ldap-iaa-providers-bundle/nifi-ldap-iaa-providers/src/main/java/org/apache/nifi/ldap/LdapProvider.java [hunk 2]
```diff
@@ -193,8 +192,41 @@ public final void onConfigured(final LoginIdentityProviderConfigurationContext c
 
         final LdapUserSearch userSearch = new FilterBasedLdapUserSearch(userSearchBase, userSearchFilter, context);
 
-        // bind
-        final BindAuthenticator authenticator = new BindAuthenticator(context);
+
+        String rawAuthenticatorType = configurationContext.getProperty("Authenticator Type");
+        AuthenticatorType authenticatorType;
+
+        if (StringUtils.isBlank(rawAuthenticatorType))
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-ldap-iaa-providers-bundle/nifi-ldap-iaa-providers/src/main/java/org/apache/nifi/ldap/LdapProvider.java]: Similarly to above, braces are encouraged to be on the same line as control statements to reduce code height. You can download the checkstyle rules for your IDE and have them automatically applied.

**Comment by alopresto** [comment]
Please start your commit messages with the Jira ticket number you are working on. This helps reviewers and the community track the code. 

Example: "NIFI-4293 Added nifi-ldap-iaa support for PasswordComparisonAuthenticator".

**Comment by alopresto** [comment]
I would suggest adding user notes and clear documentation to ensure users are aware that while `SHA-1` and salted SHA-1 (`SSHA`) hashing are supported for compatibility with existing systems, [RFC 2307](https://tools.ietf.org/html/rfc2307#section-7) notes that these algorithms are old, and [SHA-1 has seen collisions in the wild](https://shattered.io/). 

There is [extended support](https://www.redpill-linpro.com/techblog/2016/08/16/ldap-password-hash.html) using the OS `crypt` module for stronger hashing algorithms like `SHA-512` to be used with additional rounds, effectively creating a key stretching algorithm similar to PBKDF2. 

Additional reference: [OpenLDAP FAQ](http://www.openldap.org/faq/data/cache/347.html)

**Comment by joewitt** [comment]
closing due to inactivity

**Pull request closed without merging**